lib/pure/strutils.nim

```diff
@@ -2050,97 +2050,145 @@ func contains*(s: string, chars: set[char]): bool =
   ## * `find func<#find,string,set[char],Natural,int>`_
   return find(s, chars) >= 0
 
-func replace*(s, sub: string, by = ""): string {.rtl,
+func replace*(s: string, sub, by: char, maxOccurrence = -1): string {.rtl,
```

Comment:
`s/maxOccurrence/max/`  (maxOccurrence is too long to type/remember when used as optional param)

and max is just as clear